src/Utilities/SafeMethods.php

```diff
@@ -43,16 +65,40 @@ public function file_put_contents(
         int $flags = self::FILE_PUT_CONTENTS_FLAGS,
         $context = null
     ): int {
-        return \Safe\file_put_contents(
-            $filename,
-            $data,
-            $flags,
-            $context
-        );
+        error_clear_last();
+        if (null !== $context) {
+            $result = \file_put_contents($filename, $data, $flags, $context);
+        } else {
+            $result = \file_put_contents($filename, $data, $flags);
+        }
+        if (false === $result) {
+            throw FilesystemException::createFromPhpError();
+        }
+
+        return $result;
     }
 
-    public function file_get_contents(string $filename): string
-    {
-        return \Safe\file_get_contents($filename);
+    public function file_get_contents(
+        string $filename,
+        bool $use_include_path = false,
+        $context = null,
+        int $offset = 0,
+        int $maxlen = null
+    ): string {
+        error_clear_last();
+        if (null !== $maxlen) {
+            $result = \file_get_contents($filename, $use_include_path, $context, $offset, $maxlen);
+        } elseif (0 !== $offset) {
+            $result = \file_get_contents($filename, $use_include_path, $context, $offset);
+        } elseif (null !== $context) {
+            $result = \file_get_contents($filename, $use_include_path, $context);
+        } else {
+            $result = \file_get_contents($filename, $use_include_path);
+        }
+        if (false === $result) {
+            throw FilesystemException::createFromPhpError();
```

Comment:
The implementation of this exception requires you to do do a `error_clear_last` before, this is an implementation detail the called should not be aware of, it's leaking responsibilities. 

I think the same as with the other ones, i'd just make that exception get 2 strings in the constructor, and do all that 'magic' here.

```php
public function file_get_contents(..): string {
   error_clear_last();

   [..]

   if (false === $result) {
            $error = error_get_last();
            throw FileSystemException($error['message'] ?? 'An error occured', 0, $error['type'] ?? 1);
   }
}
```

that way it's al nicely done in 1 method.

To make it more explicit you could do:

```php
public function file_get_contents(..): string {
   error_clear_last();

   $result = $this->nativeFileGetContents(...);
   if (false === $result) {
            $error = error_get_last();
            throw FileSystemException($error['message'] ?? 'An error occured', 0, $error['type'] ?? 1);
   }

  return $result;
}

private function nativeFileGetContents(..): mixed {
        if (null !== $maxlen) {
            return \file_get_contents($filename, $use_include_path, $context, $offset, $maxlen);
        } elseif (0 !== $offset) {
            return \file_get_contents($filename, $use_include_path, $context, $offset);
        } elseif (null !== $context) {
            return \file_get_contents($filename, $use_include_path, $context);
        } else {
            return \file_get_contents($filename, $use_include_path);
        }
}
```